webapp/components/reftest-analyzer.js

```diff
@@ -174,6 +180,9 @@ class ReftestAnalyzer extends LoadingState(PolymerElement) {
 
     // Set the img srcs manually so that we can promisify them being loaded.
     const imagePromises = ['before', 'after'].map(prop => {
+      if (!this[prop]) {
+        return Promise.reject(`${prop} is empty`);
```

Comment:
I sent https://github.com/web-platform-tests/wpt.fyi/pull/1354 to simplify this a bit. When a function returns a promise in two places or does a lot of work outside the promified paths I always wonder if it's possible for it to throw an exception instead of rejecting, and just having a single promise with all the work inside makes it obvious.